phalcon/mvc/model/binder.zep

```diff
@@ -0,0 +1,197 @@
+	 */
+	public function getCache() -> <BackendInterface>
+	{
+		return this->cache;
+	}
+
+	/**
+	 * Get params classes from cache by key
+	 */
+	protected function getParamsFromCache(string cacheKey) -> array|null
+	{
+		var cache, internalParams;
+
+		if fetch internalParams, this->internalCache[cacheKey] {
+			return internalParams;
+		}
+
+		let cache = this->cache;
+
+		if cache != null && cache->exists(cacheKey) {
+			return cache->get(cacheKey);
+		}
+
+		return null;
+	}
+
+	/**
+	 * Get modified params for handler using reflection
+	 */
+	protected function getParamsFromReflection(object handler, array params, string cacheKey, var methodName) -> array
+	{
+		var methodParams, reflection, paramKey, methodParam, paramsCache, className, realClasses = null,
+			boundModel, cache, handlerClass, reflectionClass;
+		let paramsCache = [];
+		if methodName != null {
+			let reflection = new \ReflectionMethod(handler, methodName);
+		} else {
+			let reflection = new \ReflectionFunction(handler);
+		}
+
+		let cache = this->cache;
+
+		let methodParams = reflection->getParameters();
+		for paramKey, methodParam in methodParams {
+			let reflectionClass = methodParam->getClass();
+			if reflectionClass {
+				let className = reflectionClass->getName();
+				if typeof className == "string" {
+					let boundModel = null;
+					if className == "Phalcon\\Mvc\\Model" {
+						if realClasses == null {
+							if handler instanceof BindModelInterface {
+								let handlerClass = get_class(handler);
+                        		let realClasses = {handlerClass}::getModelName();
+                        	} elseif handler instanceof BindableInterface {
+                        		let realClasses = handler->getModelName();
+                        	} else {
+                        		throw new Exception("Handler must implement Phalcon\\Mvc\\Model\\Binder\\BindableInterface in order to use Phalcon\\Mvc\\Model as parameter");
+                        	}
+						}
+						if typeof realClasses == "array" {
+							if fetch className, realClasses[paramKey] {
+								let boundModel = {className}::findFirst(params[paramKey]);
+							} else {
+								throw new Exception("You should provide model class name for ".paramKey." parameter");
+							}
+						} elseif typeof realClasses == "string" {
+							let boundModel = {realClasses}::findFirst(params[paramKey]);
+							let className = realClasses;
+						} else {
+							throw new Exception("getModelName should return array or string");
+						}
+					} elseif is_subclass_of(className, "Phalcon\\Mvc\\Model") {
+						let boundModel = {className}::findFirst(params[paramKey]);
+					}
+
+					if boundModel != null {
+						let params[paramKey] = boundModel;
+                        let this->boundModels[paramKey] = boundModel;
+                        if cache != null {
+                        	let paramsCache[paramKey] = className;
+                        }
+					}
+				}
+			}
+		}
```

Comment:
How about this?:

```php
for paramKey, methodParam in methodParams {
	let reflectionClass = methodParam->getClass();

	if !reflectionClass {
		continue;
	}

	let className = reflectionClass->getName();

	if typeof className != "string" {
		continue;
	}

	// ...
}
```

Also, what value could [`reflectionClass->getName()`](http://php.net/manual/en/reflectionclass.getname.php) be if it's not a string?